html/cross-origin-opener-policy/resources/coop-coep.py

```diff
@@ -21,13 +21,41 @@ def main(request, response):
 <script src="/common/get-host-info.sub.js"></script>
 <iframe></iframe>
 <script>
-  const navigate = new URL(location).searchParams.get("navigate");
-  if (navigate !== null) {
-    self.location = navigate;
+  const params = new URL(location).searchParams;
+  const navHistory = params.get("navHistory");
+  const avoidBackAndForth = params.get("avoidBackAndForth");
+  const navigate = params.get("navigate");
+  const openerDOMAccess = params.get("openerDOMAccess");
+  // Need to wait until the page is fully loaded before navigating
+  // so that it creates a history entry properly.
+  async function fullyLoaded() {
```

Comment:
This can be a synchronous function. Or maybe better, just get rid of the function and store the Promise value directly.

Reply:
First I wanted to use `await fullyLoaded()` but then recalled that top-level `await` can't be used, so resorted to using `.then`. But feel free to tweak this 🙂